### ISSUE TITLE
golang.org/s: print an index at the root

### ISSUE THREAD

**Issue by robpike** [golang/go#43676]
The URL shortener for the golang.org web site, golang.org/s/ contains useful historical links, but I cannot discover a way to see the list. Both golang.org/s and golang.org/s/index produce

`
not found
`

A simple feature request: golang.org/s could list the entries.

**Comment by changkun** [comment]
I was interested in all `/s` links, but could not find a way to view them. So I searched these places: 1) the source code 2) all commit messages 3) golang-nuts 4) golang-dev 5) wiki 6) google and summarized most of them as a list at [golang.design/history](https://github.com/golang-design/history).

People who interested in this, can search from the [README](https://raw.githubusercontent.com/golang-design/history/master/README.md) document via `golang.org/s`, here are some of them (not sure if there are more):

```
https://golang.org/s/semicolon-proposal
https://golang.org/s/go1
https://golang.org/s/go11sched
https://golang.org/s/go11func
https://golang.org/s/go11return
https://golang.org/s/go12nil
https://golang.org/s/go12slice
https://golang.org/s/go12symtab
https://golang.org/s/go12encoding
https://golang.org/s/go13todo
https://golang.org/s/go13linker
https://golang.org/s/go13heapdump
https://golang.org/s/go13compiler
https://golang.org/s/go13nacl
https://golang.org/s/go14android
https://golang.org/s/go14internal
https://golang.org/s/go14nopkg
https://golang.org/s/go14gc
https://golang.org/s/go14customimport
https://golang.org/s/go15vendor
https://golang.org/s/go15bootstrap
https://golang.org/s/go15gomaxprocs
https://golang.org/s/go15gcpacing
https://golang.org/s/go15trace
https://golang.org/s/go17ssa
https://golang.org/s/go19inliningtalk
https://golang.org/s/generatedcode
https://golang.org/s/gctoc
https://golang.org/s/go2designs
https://golang.org/s/dev.cc
https://golang.org/s/execmodes
https://golang.org/s/better-linker
https://golang.org/s/win2000-golang-nuts
https://golang.org/s/contigstacks
https://golang.org/s/types-tutorial
https://golang.org/s/using-guru
https://golang.org/s/stdwhy
https://golang.org/s/oracle-design
https://golang.org/s/oracle-user-manual
https://golang.org/s/cgihttpproxy
https://golang.org/s/sqldrivers
https://golang.org/s/builderplan
```

**Comment by ALTree** [comment]
@changkun here's one you're missing:

- https://golang.org/s/owners

**Comment by robpike** [comment]
@changkun Thanks very much! But I'd still like the site to offer its own, automatically maintained index.

**Comment by changkun** [comment]
We would love that feature too :)

**Comment by rsc** [comment]
I can see the full index, and it's kind of full of junk.
I'm not convinced that publishing the full table of junk is worthwhile.
Even worse, there are clearly links there that were created assuming
that the full names and links would not be made into a public table.
(You can argue that was a mistake, but they exist.)

If we are going to publish an index on /s, there should be a separate
"public bool" bit stored with each link that must be set explicitly
for a link to appear in the index.

Please do NOT do this, at least not without adding that bit.

**Comment by changkun** [comment]
Thanks for the clarification about the junk and privately created links. Then I don't think that the community can do anything about this as the sent CL was done blindly by just reading the code and marking public bool would require admin permission to at least view them all. Still looking forward to the feature be launched in the future as reason commented previously.

**Comment by robpike** [comment]
I have made this a proposal to try to force discussion and closure.

**Comment by rsc** [comment]
I don't think we really need a proposal. I am happy to add this feature as long as there is a public/private bit too.
I'm already working on golang.org recently. If you leave it for me, I will implement it.
